server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -138,6 +138,43 @@ public void testParseWithShadowedMultiField() throws Exception {
         assertNotNull(doc.rootDoc().getField("field.keyword"));
     }
 
+    public void testRuntimeFieldDoesNotShadowObjectChildren() throws IOException {
```

Comment:
Should we expand testing for arrays, both arrays of leaves and arrays of objects?

Reply:
++, will add some more tests here

Reply:
Shall we also have a test for arrays that hold leaf values instead of objects? What I am after is verifying that using getLeafMapper is the right choice in all cases when parsing an array.